sdk/quantum/Azure.Quantum.Jobs/tests/QuantumJobClientTestEnvironment.cs

```diff
@@ -1,12 +1,126 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
 using Azure.Core.TestFramework;
+using Azure.Identity;
+using Newtonsoft.Json;
 
 namespace Azure.Quantum.Jobs.Tests
 {
     public class QuantumJobClientTestEnvironment : TestEnvironment
     {
+        private bool _initialized = false;
+        private readonly Lazy<AzLoginAccessTokenInfo> _azLoginAccessTokenInfo = new Lazy<AzLoginAccessTokenInfo>(QuantumJobClientTestEnvironment.GetAzLoginAccessTokenInfo);
+
         public string WorkspaceName => GetRecordedVariable("WORKSPACE_NAME");
+
+        public string GetRandomId(string idName)
+        {
+            var randomId = Guid.NewGuid().ToString("N");
+            var randomIdVariableName = $"RANDOM_ID_{idName}";
+            if (Mode == RecordedTestMode.Record)
+            {
+                Environment.SetEnvironmentVariable(randomIdVariableName, randomId);
+            }
+            return GetRecordedVariable(randomIdVariableName);
+        }
+
+        public new TokenCredential Credential
+        {
+            get
+            {
+                if (Mode == RecordedTestMode.Record)
+                {
+                    return new AzureCliCredential();
```

Comment:
This doesn't look right.  We can record as part of the CI, etc., which I don't believe has setup the CLI for auth.